factoryWithTypeCheckers.js

```diff
@@ -206,7 +206,8 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
           }
         }
       }
-      if (props[propName] == null) {
+      // undefined == null and null == null are true
+      if (props[propName] === null || props[propName] === undefined) {
```

Comment:
why is this a better change? `== null` is more reliable, since `undefined` can be redefined.

Reply:
> why is this a better change? `== null` is more reliable, since `undefined` can be redefined.

Thanks @ljharb I found this question on stackoverflow, it looks like a problem related to legacy browsers only, but your review did makes sense.

Reply:
Also according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined), undefined should be un-writable.